.github/workflows/docsite.yml

```diff
@@ -0,0 +1,49 @@
+name: Build and Publish docsite
+
+on:
+  pull_request:
+    branches:
+      - master
+  push:
+    branches:
+      - master
+
+jobs:
+  build:
+    name: Build Mkdocs website
+    runs-on: ubuntu-latest
+    strategy:
+        max-parallel: 1
+    container:
+      image: squidfunk/mkdocs-material
+    steps: 
+      - name: Checkout
+        uses: actions/checkout@v2.3.1
+      - name: Build
+        run: mkdocs build -s
+      - name: Upload artifact
+        uses: actions/upload-artifact@v1
+        with:
+          name: site
+          path: site
+
+  publish:
+    needs: build
+    name: Publish Mkdocs website to GH Pages
+    runs-on: ubuntu-latest
+    strategy:
+      max-parallel: 1
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2.3.1
+      - name: Download artifact
+        uses: actions/download-artifact@v1
+        with:
+          name: site
+          path: site
+      - name: Publish
+        uses: JamesIves/github-pages-deploy-action@4.1.7
+        with:
+          branch: mkdocs
+          folder: site
+          token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Judging by the documentation the token shouldn't need to be passed directly, so we should be okay if we just remove this line.

```suggestion
```